lib/y2logsstep.pm

```diff
@@ -232,17 +233,25 @@ sub verify_license_translations {
 }
 
 sub save_upload_y2logs {
-    my ($self, $suffix) = @_;
-    $suffix //= '';
+    my ($self, %args) = @_;
 
     return if (get_var('NOLOGS'));
-
-    assert_script_run 'sed -i \'s/^tar \(.*$\)/tar --warning=no-file-changed -\1 || true/\' /usr/sbin/save_y2logs';
-    my $filename = "/tmp/y2logs$suffix.tar" . get_available_compression();
-    assert_script_run "save_y2logs $filename", 180;
-    upload_logs $filename;
-    save_screenshot();
-    $self->investigate_yast2_failure();
+    $args{suffix} //= '';
+
+    # Try to recover network if cannot reach gw and upload logs if everything works
+    # Do not test/recover network if collect from installation system, as it won't work anyway with current approach
+    if (can_upload_logs() || recover_network(skip_recovery => $args{no_ntwrk_recovery})) {
```

Comment:
not big thing but maybe you will rename `no_ntwrk_recovery` to `skip_recovery` ?

Reply:
next thing which I realized is that in case `skip_recovery=true` you should not return results of `can_upload_logs` . it does not make sense , than I understand that actually `recover_network` function don't need this variable at all. We can do it simpler :
```
my $connection = can_upload_logs();
 if ( $connection ||  $args{no_ntwrk_recovery} || recover_network()) {
if($connection) {
 assert_script_run 'sed -i \'s/^tar \(.*$\)/tar --warning=no-file-changed -\1 || true/\' /usr/sbin/save_y2logs';
        my $filename = "/tmp/y2logs$args{suffix}.tar" . get_available_compression();
        assert_script_run "save_y2logs $filename", 180;
        upload_logs $filename;
}
        save_screenshot();
        $self->investigate_yast2_failure();
}
```

Reply:
I leave `no_ntwrk_recovery` with old name to not confuse you even more , but my first comment remains valid too

Reply:
It was already like that, and after your comments I started to pass parameter to the `recover_network`. So I've changed it back this I also believe is more clear way.